javascript/index.js

```diff
@@ -4,37 +4,143 @@
 // 🚨🚨🚨 Comment out the below code before you start working on the code
 
 // Out of sync
-  getInstruction("mashedPotatoes", 0, (step1) => {
+getInstruction(
+  "mashedPotatoes",
+  0,
+  (step1) => {
     document.querySelector("#mashedPotatoes").innerHTML += `<li>${step1}</li>`;
-  }, (error) => console.log(error));
-  
-  getInstruction("mashedPotatoes", 1, (step2) => {
+  },
+  (error) => console.log(error)
+);
+
+getInstruction(
+  "mashedPotatoes",
+  1,
+  (step2) => {
     document.querySelector("#mashedPotatoes").innerHTML += `<li>${step2}</li>`;
-  }, (error) => console.log(error));
-  
-  getInstruction("mashedPotatoes", 2, (step3) => {
+  },
+  (error) => console.log(error)
+);
+
+getInstruction(
+  "mashedPotatoes",
+  2,
+  (step3) => {
     document.querySelector("#mashedPotatoes").innerHTML += `<li>${step3}</li>`;
-  }, (error) => console.log(error));
-  
-  getInstruction("mashedPotatoes", 3, (step4) => {
+  },
+  (error) => console.log(error)
+);
+
+getInstruction(
+  "mashedPotatoes",
+  3,
+  (step4) => {
     document.querySelector("#mashedPotatoes").innerHTML += `<li>${step4}</li>`;
-  }, (error) => console.log(error));
-  
-  getInstruction("mashedPotatoes", 4, (step5) => {
+  },
+  (error) => console.log(error)
+);
+
+getInstruction(
+  "mashedPotatoes",
+  4,
+  (step5) => {
     document.querySelector("#mashedPotatoes").innerHTML += `<li>${step5}</li>`;
     document.querySelector("#mashedPotatoesImg").removeAttribute("hidden");
-  }, (error) => console.log(error));
-
-
+  },
+  (error) => console.log(error)
+);
 
 // Iteration 1 - using callbacks
 // ...
 
-// Iteration 2 - using promises
-// ...
+// Iteration 2 - using promises + image (Bonus 1)
 
-// Iteration 3 using async/await
-// ...
+obtainInstruction("steak", 0)
```

Comment:
this could be shortened by using recursion. 
something like this:
```js
function cookRecipe(recipe, maxSteps, currentStep) {
  if (currentStep >= maxSteps) return;

  obtainInstruction(recipe, currentStep).then((result) => {
    document.querySelector("#" + recipe).innerHTML += `<li>${result}</li>`;
    cookRecipe(recipe, maxSteps, currentStep + 1);
  });
}

cookRecipe("steak", 8, 0);
```

Reply:
the recursive part is that in the .then it calls cookRecipe again :) but incrementing the step by 1 each time it does.

Reply:
you could even use the error handling to your advantage and remove the if (currentStep > maxSteps) at the beginning like so: 
```js
function cookRecipe(recipe, step = 0) {
  obtainInstruction(recipe, step)
    .then((result) => {
      document.querySelector("#" + recipe).innerHTML += `<li>${result}</li>`;
      cookRecipe(recipe, step + 1);
    })
    .catch((err) => console.log(err));
}

cookRecipe("steak");
```